contrib/gofiber/fiber.v2/fiber_test.go

```diff
@@ -158,6 +158,34 @@ func TestCustomError(t *testing.T) {
 	assert.Equal(fiber.ErrBadRequest, span.Tag(ext.Error).(*fiber.Error))
 }
 
+func TestUserContext(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	// setup
+	router := fiber.New()
+	router.Use(Middleware(WithServiceName("foobar")))
+
+	router.Get("/", func(c *fiber.Ctx) error {
+		// check if not default empty context
+		assert.NotEmpty(c.UserContext())
+		span, _ := tracer.StartSpanFromContext(c.UserContext(), "http.request")
+		defer span.Finish()
+		return c.SendString("test")
+	})
+	r := httptest.NewRequest("GET", "/", nil)
+
+	router.Test(r, 100)
+
+	// verify both middleware span and router span finished
+	spans := mt.FinishedSpans()
+	assert.Equal(spans[1].SpanID(), spans[0].ParentID())
+	if len(spans) < 2 {
+		t.Fatalf("no spans")
+	}
+}
```

Comment:
Let's use `assert.Len(spans, 2)` instead of an if, and do this *before* the `assert.Equal`. 
Putting this here does nothing since if `len(spans) < 2`, `spans[1]` will cause a panic above and we will never reach this code.